gateway/dataimpl/server_v1/kvserver.go

```diff
@@ -1508,8 +1538,16 @@ func (s *KvServer) GetAllReplicas(in *kv_v1.GetAllReplicasRequest, out kv_v1.KvS
 }
 
 func (s *KvServer) checkKey(key string) *status.Status {
-	if len(key) > 250 {
-		return s.errorHandler.NewKeyTooLongStatus(key)
+	if len(key) > 250 || len(key) < 1 {
+		return s.errorHandler.NewInvalidKeyLengthStatus(key)
+	}
+
+	return nil
+}
+
+func (s *KvServer) checkCAS(cas *uint64) *status.Status {
```

Comment:
Note that a 0 CAS is technically 'valid' in that it simply means that the CAS should be ignored (except in the case of Unlock, where a CAS is required always).  This change is probably correct since we've typically opted to consider such an 'ignored' value for an optional field in Protostellar to be an error (but weren't for this CAS case until this commit), but I'm just going to mention this here for @chvck to chime in on.

Reply:
I think that this sounds sensible. Cas for most operations is effectively optional so the "zero value" is nil rather than 0. Setting actively the cas to 0 seems like a error case to me.